geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/legacy/z3/Z3WritableIndex.scala

```diff
@@ -37,14 +37,15 @@ trait Z3WritableIndex extends AccumuloFeatureIndex {
 
   def hasSplits: Boolean
 
-  override def delete(sft: SimpleFeatureType, ds: AccumuloDataStore, shared: Boolean): Unit = {
-    val table = getTableName(sft.getTypeName, ds)
-    if (ds.tableOps.exists(table)) {
-      // we need to synchronize deleting of tables in mock accumulo as it's not thread safe
-      if (ds.connector.isInstanceOf[MockConnector]) {
-        ds.connector.synchronized(ds.tableOps.delete(table))
-      } else {
-        ds.tableOps.delete(table)
+  override def delete(sft: SimpleFeatureType, ds: AccumuloDataStore, partition: Option[String]): Unit = {
+    getTableNames(sft, ds, partition).par.foreach { table =>
```

Comment:
.par has some parallelism settings implicitly set.  Do we want to surface those in any configurable way?

Reply:
i don't think it's worth it... from my understanding you can just use a different thread pool than the default?